portfolio/src/main/webapp/script.js

```diff
@@ -31,23 +31,186 @@ function addRandomFunFact() {
+                + '</td></tr>';
+        }
+
+        commentTableText += '</table>';
+
+        commentsContainer.innerHTML = commentTableText;
+
+        commentsAdded = 1;
+    }
+}
+
+function displayCommentBoxes(sectionId){
+    //if visible, hide and change button to "show"
+    //if hidden, show and change button to "hide"
+    if(document.getElementById("forms-wrapper-" + sectionId).style.display == "block"){
+        document.getElementById("forms-wrapper-" + sectionId).style.display = "none";
+        document.getElementById(sectionId + "-display-button").innerHTML = "Post a Comment";
+    }else if(document.getElementById("forms-wrapper-" + sectionId).style.display != "block"){
+        document.getElementById("forms-wrapper-" + sectionId).style.display = "block";
+        document.getElementById(sectionId + "-display-button").innerHTML = "Hide Post Comment Boxes";
+    }
+
+}
+
+function getNumComments(myLocation){
+
+    numComments = document.getElementById("num-comments-" + myLocation).value;
+
+    return numComments;
+
+}
+
+//Used in getComments, to make sure the num comments is up to date
+function refreshNumComments(myLocation){
+   
+    var globalNumComments = document.getElementById("num-comments-" + myLocation).value;
+
+    const updateNumComments = fetch('/comments?num-comments=' + globalNumComments + "&location=" + myLocation, { method: 'PUT'});
+}
+
+//Used in updating num comments, when the number should change and the comments should be hidden.
+function updateNumComments(myLocation){
+
+    refreshNumComments(myLocation);
+
+    hideComments(myLocation);
+
+}
+
+function hideComments(myLocation){
+    document.getElementById('comments-container-' + myLocation).innerText = "";
+    //Indicate that the comments are no longer added
+    commentsAdded = 0;
+    commentTableText = `<table id="comment-table"> 
+            <tr> 
+                <th>Comments</th> 
+            </tr>`;
+    
+    //Change innerHTML of the get button here because hideComments is called in updating the number of comments
+    //to display, and we want to hide comments after changing the number of comments
+    document.getElementById(myLocation + "-get-button").innerHTML = "Display Comments";
+    shouldDisplay = true;
+
+}
+
+function deleteComments(myLocation){
+    console.log('Deleting comments.');
+    
+    //Show a pop-up confirm box so the user doesn't accidentally delete all comments without confirmation
+    if (confirm("By clicking 'OK', you will delete all comments.")) {
+        // The fetch() function returns a Promise because the request is asynchronous.
+        const responseDeletePromise = fetch('/delete-data?location=' + myLocation, { method: 'POST'});
+
+        //Call the function to fetch comments from the server so that the now-deleted comments are removed from the page
+        responseDeletePromise.then(getComments);
+
+        location.reload();
```

Comment:
where is this "location" defined?

Reply:
location.reload() is a predefined JS function for a predefined JS class named location. I'll add a comment.